client/my-sites/sharing/connections/mailchimp-settings.jsx

```diff
@@ -0,0 +1,73 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { requestSettingsUpdate } from 'state/mailchimp/settings/actions';
+import QueryMailchimpLists from 'components/data/query-mailchimp-lists';
+import QueryMailchimpSettings from 'components/data/query-mailchimp-settings';
+import FormRadio from 'components/forms/form-radio';
+import FormLabel from 'components/forms/form-label';
+import FormFieldset from 'components/forms/form-fieldset';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+const MailchimpSettings = ( {
+	siteId,
+	keyringConnections,
+	requestSettingsUpdateAction,
+	mailchimpLists,
+	mailchimpListId,
+} ) => {
+	const chooseMailchimpList = listId => {
+		requestSettingsUpdateAction( siteId, {
+			follower_list_id: listId,
+			keyring_id: keyringConnections[ 0 ].ID,
+		} );
+	};
+
+	return (
+		<div>
+			<QueryMailchimpLists siteId={ siteId } />
+			<QueryMailchimpSettings siteId={ siteId } />
+			<FormFieldset>
+				{ mailchimpLists.map( list => (
+					<FormLabel key={ list.id }>
+						<FormRadio
+							value={ list.id }
+							checked={ list.id === mailchimpListId }
+							onChange={ () => chooseMailchimpList( list.id ) }
```

Comment:
This creates a new handler function for each list item on each rerender. That can be somewhat inefficient. It could be a class method that reads the `list.id` from the `event.target`:
```js
chooseMailchimpList: event => {
  const listId = event.target.value;
  this.props.requestSettingsUpdate( listId, this.props.siteId, ... );
}
```
Then the're ever only one instance of that functions for the whole lifetime of the component.